src/Api/PubnubApi/NewtonsoftJsonDotNet.cs

```diff
@@ -1234,14 +1234,19 @@ public virtual T DeserializeToObject<T>(List<object> listObject)
 
         public Dictionary<string, object> DeserializeToDictionaryOfObject(string jsonString)
         {
+            Dictionary<string, object> result = null;
             try
             {
-                return JsonConvert.DeserializeObject<Dictionary<string, object>>(jsonString);
+                if (JsonFastCheck(jsonString))
+                {
+                    result = JsonConvert.DeserializeObject<Dictionary<string, object>>(jsonString);
+                }
             }
             catch
             {
-                return null;
+                //ignore
```

Comment:
I'd leave the `return null` intact just in case we receive a malformed JSON. The deserialization may still fail, as the JsonFastCheck function can tell that its payload is invalid, but cannot prove that it's valid.

Reply:
Since we have `Dictionary<string, object> result = null;`, any exception will return `null` because of `return result;`
So for malformed JSON, null will be returned.

Reply:
Right, agreed.